src/OpenSpaceToolkit/Mathematics/Geometry/3D/Objects/Point.cpp

```diff
@@ -52,7 +52,9 @@ bool                            Point::operator ==                          (
         return false ;
     }
 
-    return (x_ == aPoint.x_) && (y_ == aPoint.y_) && (z_ == aPoint.z_) ;
+    Real tolerance = 1e-15 ;
+
+    return (x_.isNear(aPoint.x_, tolerance)) && (y_.isNear(aPoint.y_, tolerance)) && (z_.isNear(aPoint.z_, tolerance)) ;
```

Comment:
Looks like this diff was added to this PR by mistake?

Reply:
So looks like in Math I'm seeing slightly different behaviour when using -O0  flag:
 -- Point -------------------------------------------------------------------------------------------
    X:                                       4.6541204004284447e-310                  
    Y:                                       0.0                                      
    Z:                                       3.0                                      
----------------------------------------------------------------------------------------------------
Since it's not exactly 0, it thinks it's a different point from (0.0, 0.0, 3.0). Which is then causing the test to fail as it returns a PointSet instead of a Point. I've fixed this in the commit, but it feels kind of annoying to fix something for the debug flag... though maybe this could happen in prod as well. In general I feel like checking equality of doubles is iffy... Thoughts?

The unit tests were previously failing cause of this.

(Sorry I posted this on slack but didn't tag you)

Reply:
(The -O0 flag is applied when we `make test-coverage-app` which is the command used in the gitlab CI)

Reply:
Any chance we could modify the faulty test to use the `Point::isNear` method instead? I think the == operator should still assert strict equality, because sometimes we actually want to check for strict binary equality (which I agree shouldn't be used in practice because of float comparison)

Reply:
That's fair yeah. I can update the test. Just wanted to make sure we wanted to stick with this behaviour. Thanks Lucas.